src/plugins/console/public/lib/kb/kb.js

```diff
@@ -27,74 +24,38 @@ import Api from './api';
 let ACTIVE_API = new Api();
 const isNotAnIndexName = (name) => name[0] === '_' && name !== '_all';
 
-const idAutocompleteComponentFactory = (name, parent) => {
-  return new IdAutocompleteComponent(name, parent);
-};
 const parametrizedComponentFactories = {
   getComponent: function (name, parent, provideDefault) {
     if (this[name]) {
       return this[name];
     } else if (provideDefault) {
-      return idAutocompleteComponentFactory;
+      return new ListComponent(name, [], parent, false);
     }
   },
   index: function (name, parent) {
-    if (isNotAnIndexName(name)) return;
-    return new IndexAutocompleteComponent(name, parent, false);
-  },
-  indices: function (name, parent) {
     if (isNotAnIndexName(name)) return;
     return new IndexAutocompleteComponent(name, parent, true);
   },
-  type: function (name, parent) {
-    return new TypeAutocompleteComponent(name, parent, false);
-  },
-  types: function (name, parent) {
-    return new TypeAutocompleteComponent(name, parent, true);
-  },
-  id: function (name, parent) {
-    return idAutocompleteComponentFactory(name, parent);
-  },
-  transform_id: function (name, parent) {
-    return idAutocompleteComponentFactory(name, parent);
-  },
-  username: function (name, parent) {
-    return new UsernameAutocompleteComponent(name, parent);
-  },
-  user: function (name, parent) {
-    return new UsernameAutocompleteComponent(name, parent);
-  },
-  template: function (name, parent) {
-    return new LegacyTemplateAutocompleteComponent(name, parent);
-  },
-  task_id: function (name, parent) {
-    return idAutocompleteComponentFactory(name, parent);
-  },
-  ids: function (name, parent) {
-    return idAutocompleteComponentFactory(name, parent, true);
-  },
   fields: function (name, parent) {
     return new FieldAutocompleteComponent(name, parent, true);
   },
   field: function (name, parent) {
     return new FieldAutocompleteComponent(name, parent, false);
   },
-  nodes: function (name, parent) {
-    return new ListComponent(
-      name,
-      ['_local', '_master', 'data:true', 'data:false', 'master:true', 'master:false'],
-      parent
-    );
-  },
-  node: function (name, parent) {
-    return new ListComponent(name, [], parent, false);
+  // legacy index templates
+  template: function (name, parent) {
+    return new LegacyTemplateAutocompleteComponent(name, parent);
   },
+  // composable index templates
+  // currently seems to be unused, but that is a useful functionality
```

Comment:
Can you elaborate on this? I believe this functionality this was added via https://github.com/elastic/kibana/pull/124655. I don't remember it being removed, so I wonder if there was a regression somewhere 🤔

Reply:
Good catch, thanks @alisonelizabeth! I checked the PR and the dynamic parameters were added there and they indeed still work correctly. But their usage was in the `generated` definitions file. The next time we used the script (https://github.com/elastic/kibana/pull/153098), we reverted the manual changes made to those definitions. So instead of `_component_template/{component_template}` we have `_component_template/{name}` because `name` is how it's defined in the specs. I'll open an issue to use those dynamic parameters, but in the `overrides` files. 
I think the goal is to not edit generated files and not even have to review them when they are re-generated using the script. On the other hand, we should spend more time on updating the overrides.

Reply:
Sounds good. Thanks for the explanation and looking into this!